oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/storage/query/IMetricsQueryDAO.java

```diff
@@ -76,5 +87,45 @@ public static IntValues sortValues(IntValues origin, List<String> expectedOrder,
             }
             return origin;
         }
+
+        /**
+         * Compose the multiple metric result based on conditions.
+         */
+        public static List<MetricsValues> composeLabelValue(final MetricsCondition condition,
+            final List<String> labels,
+            final List<String> ids,
+            final Map<String, DataTable> idMap) {
+            List<String> allLabels;
+            if (Objects.isNull(labels) || labels.size() < 1) {
+                allLabels = idMap.values().stream()
+                    .flatMap(dataTable -> dataTable.keys().stream())
+                    .distinct().collect(Collectors.toList());
+            } else {
+                allLabels = labels;
+            }
+            final int defaultValue = ValueColumnMetadata.INSTANCE.getDefaultValue(condition.getName());
```

Comment:
I feel like you should just copy the old codes rather than re-implement.

Reply:
Once you have the `allLabels`, copying is a safer way. This re-implementation is hard to be reviewed.

Reply:
Could you revert them? Is there any block there?

Reply:
I used to follow copy and past strategy but found it would bring more mess more than what I expect. Unlike Es, H2 and Influxdb don't fetch all data before composing the final result object. That causes I have to tweak the current algorithm unavoidably. Future more, there are too many iterations that bring some performance concerns, I want to optimize it in the new implement.

Reply:
I agree with that lamda is too hard to review. I'd like to improve it with other code styles.

Reply:
And I also add a unit test case to help define the behavior of this function. Please review it to check if it cover the main scope.

Reply:
Updated, open a new thread for review if necessary.